go/vumitools/tests/test_billing_worker.py

```diff
@@ -739,3 +744,67 @@ def test_outbound_message_session_length_custom_field(self):
             "outbound",
             session_created=False,
             session_metadata_field='foo')
+
+    @inlineCallbacks
+    def test_outbound_message_credit_cutoff_session(self):
+        self.billing_api = BillingApiMock(credit_cutoff=True)
+        dispatcher = yield self.get_dispatcher()
+
+        yield self.make_dispatch_outbound(
+            "outbound",
+            user_account="12345",
+            tag=("pool1", "1234"),
+            helper_metadata={},
+            session_event='new'
+            )
+
+        published_msg = self.ri_helper.get_dispatched_outbound()[0]
+        self.assertEqual(published_msg['session_event'], SESSION_CLOSE)
+        self.assertEqual(
+            published_msg['content'], dispatcher.credit_limit_message)
+
+    @inlineCallbacks
+    def test_outbound_message_credit_cutoff_message(self):
+        self.billing_api = BillingApiMock(credit_cutoff=True)
+        yield self.get_dispatcher()
+
+        yield self.make_dispatch_outbound(
+            "outbound",
+            user_account="12345",
+            tag=("pool1", "1234"),
+            helper_metadata={}
+            )
+
+        published_msgs = self.ri_helper.get_dispatched_outbound()
+        self.assertEqual(len(published_msgs), 0)
+
+    @inlineCallbacks
+    def test_inbound_message_credit_cutoff_session(self):
+        self.billing_api = BillingApiMock(credit_cutoff=True)
+        yield self.get_dispatcher
```

Comment:
I think you might be asserting on the wrong message (i.e. you might be asserting that the message you sent into the worker exists, rather than the message the worker forwarding existing).